src/main/scala/com/databricks/spark/avro/SchemaConverters.scala

```diff
@@ -95,26 +93,75 @@ private object SchemaConverters {
   }
 
   /**
-   * This function converts sparkSQL StructType into avro schema. This method uses two other
-   * converter methods in order to do the conversion.
-   */
-  private[avro] def convertStructToAvro[T](
-      structType: StructType,
-      schemaBuilder: RecordBuilder[T],
-      recordNamespace: String): T = {
-    val fieldsAssembler: FieldAssembler[T] = schemaBuilder.fields()
-    structType.fields.foreach { field =>
-      val newField = fieldsAssembler.name(field.name).`type`()
-
-      if (field.nullable) {
-        convertFieldTypeToAvro(field.dataType, newField.nullable(), field.name, recordNamespace)
-          .noDefault
-      } else {
-        convertFieldTypeToAvro(field.dataType, newField, field.name, recordNamespace)
-          .noDefault
+  * This function converts sparkSQL StructType into avro schema. This method uses the convertFieldType
+  * converter method in order to do the conversion.
+  */
+  private[avro] def convertStructToAvro(structType: StructType,
+                                        structName: String,
+                                        recordNamespace: String): Schema = {
+    val fields = structType.fields.map { structField =>
+      val fieldSchema = convertFieldType(structField.dataType, structField.name, recordNamespace)
+
+      if (structField.nullable) {
+        new Schema.Field(structField.name, optional(fieldSchema), null, JsonProperties.NULL_VALUE)
       }
+      else {
+        new Schema.Field(structField.name, fieldSchema, null, null.asInstanceOf[Object])
+      }
+    }.toList
+
+    Schema.createRecord(structName, null, recordNamespace, false, fields)
+  }
+
+  /**
+  * This function is used to construct field schemas of the avro record.
+  * The data type of each field is used to determine which field schema to create.
+  */
+  private[avro] def convertFieldType(dataType: DataType,
```

Comment:
This method calls back to `convertStructToAvro` to create nested records which have `structName`'s and namespaces